pkg/api/customization/kontainerdriver/store.go

```diff
@@ -45,9 +47,9 @@ func (s *store) Delete(apiContext *types.APIContext, schema *types.Schema, id st
 		return nil, nil
 	}
 
-	clustersWithKontainerDriver, err := s.ClusterIndexer.ByIndex(clusterByKontainerDriverKey, driver.Status.DisplayName)
+	clustersWithKontainerDriver, err := s.ClusterIndexer.ByIndex(clusterByGenericEngineConfigKey, id)
 	if err != nil {
-		return nil, errorsutil.WithMessage(err, fmt.Sprintf("error determing if kontainer driver [%s] was in use", driver.Status.DisplayName))
+		return nil, errorsutil.WithMessage(err, fmt.Sprintf("error determing if kontainer driver [%s] was in use", driver.Name))
```

Comment:
I would prefer to keep this as the display name so it's easier to tie it back. `k-blah` is not useful when looking at the UI.